lib/core/dispatchRequest.js

```diff
@@ -47,6 +47,13 @@ module.exports = function dispatchRequest(config) {
     }
   );
 
+  // Remove header where value is null
```

Comment:
Is it necessary to remove if the header is null?

Reply:
Hmm, could you elaborate? The issue this PR attempts to fix is that previously setting a header to `null` would leave it as is, which is not expected.

Reply:
> I suggest changes be made in lib/core/mergeConfig.js. As this is kind of special merge strategy for headers.

My main point is that. If did in `lib/core/mergeConfig.js`, we don't require special removing here.

Reply:
`dispatchRequest` currently composes the final headers-object (see [these lines](https://github.com/axios/axios/blob/bb4ca113a659bf7e79432dc2b4c1429b0dbb96d8/lib/core/dispatchRequest.js#L36-L48)), so the null-check must be done after that composition, I think.